Lib/pathlib.py

```diff
@@ -114,6 +116,16 @@ def _select_from(self, parent_path, is_dir, exists, scandir):
         yield parent_path
 
 
+class _ParentSelector(_Selector):
+    def __init__(self, name, child_parts, flavour):
+        _Selector.__init__(self, child_parts, flavour)
+
+    def _select_from(self,  parent_path, is_dir, exists, scandir):
+        path = parent_path._make_child_relpath('..')
+        for p in self.successor._select_from(path, is_dir, exists, scandir):
+            yield p
```

Comment:
It looks like all other `_Selector` classes wrap their loops in a `try`/`except PermissionError` block. Do we need to do that here as well?

Relatedly: I tried removing the `try/except` block from `_PreciseSelector._select_from` and `_WildcardSelector._select_from`, and no tests failed -- seems like we might be somewhat lacking in test coverage there :)

```suggestion
        try:
            path = parent_path._make_child_relpath('..')
            for p in self.successor._select_from(path, is_dir, exists, scandir):
                yield p
        except PermissionError:
            return
```

Reply:
We don't need try/except here because we're not calling filesystem-accesing APIs. It's needed in `_PreciseSelector` because it calls `Path.exists()` or `Path.is_dir()`, which may raise.

The handling of `PermissionError` is pretty wrong. I'll put together an issue and a patch (including tests!)

Reply:
Ahhh gotcha. So it's the `is_dir()` calls we're guarding against. Thanks!